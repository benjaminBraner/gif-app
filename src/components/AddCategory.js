import React, { useState } from 'react';
import PropTypes from 'prop-types';
import '../styles/components/AddCategory.css';

export const AddCategory = ({ setCategories }) => {


     const [inputValue, setInputValue] = useState('');

     const handleInputChange = (e) => {
          setInputValue(e.target.value)
     }

     const handleSubmit = (e) => {
          e.preventDefault();

          if (inputValue.trim().length > 2) {
               setCategories(categories => {
                    let y = [...categories].filter(c => c !== inputValue)
                    return [inputValue, ...y];
               });

               setInputValue('');
          }
     }

     return (
          <>
               <form onSubmit={handleSubmit}>
                    <input
                         type='text'
                         value={inputValue}
                         onChange={handleInputChange}
                    />
                    <button type='submit'>Submit</button>
               </form>
          </>
     )
}

AddCategory.propTypes = {
     setCategories: PropTypes.func.isRequired
}