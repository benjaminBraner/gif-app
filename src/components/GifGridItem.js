import PropTypes from "prop-types";

export const GifGridItem = ({title, url}) => {
   // console.log({id,title,url});
   return (
      <div className="card animate__animated animate__bounce">
         <p>{title}</p>
         <img src={url} alt={title}></img>
      </div>
   )             
}

GifGridItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired
}