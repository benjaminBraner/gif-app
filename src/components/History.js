import '../styles/components/History.css';


export const History = ({ histItem, setCategories }) => {

   const showElement = () => {
      setCategories(categories => {
         let y = [...categories].filter(c => c !== histItem);
         return [histItem, ...y];
      });
   }

   return (
      <>
      {histItem ? <li onClick={showElement}>{histItem}</li> : null}

      
      
      </>
   )
}
