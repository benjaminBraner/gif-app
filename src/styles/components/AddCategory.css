input {
   max-width: 500px;
   margin: 10px 5px 10px 10px;
   padding: 6px;
   font-size: 17px;

}

button {
   background-color: rgb(60, 145, 241);
   border-radius: 5px;
   border: none;
   color: #fff;
   padding: 6px;
   margin: 10px 10px 10px 5px;
   font-size: 17px;
   cursor: pointer;
}

button:hover {
   background-color: rgb(17, 97, 189);
}