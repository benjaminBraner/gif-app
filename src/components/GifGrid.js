import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";


export const GifGrid = ({ category }) => {
   const { data: images, loading } = useFetchGifs(category);

   return (
      <>
         <h3>{category}</h3>

         {loading && <p>Loading...</p>}
         {/* {loading ? <p>Loading...</p> : null} */}

         <div className="card-grid">
            {
               images.map(img => (
                  <GifGridItem
                     key={img.id}
                     {...img}
                  />
               ))
            }


         </div>
      </>
   )
};

GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}