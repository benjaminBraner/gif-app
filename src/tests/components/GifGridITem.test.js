import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem/>', () => {

   const title = "some title";
   const url = "https://someurl";
   const wrapper = shallow(<GifGridItem title={title} url={url} />);


   test('debe mostrar el componente correctamente', () => {

      expect(wrapper).toMatchSnapshot();
   })


   test('debe tener un parrafo con el title', () => {
      //estoy testeando si los argumentos que tengo en mis props se estan usando en el lugar donde yo espero
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);

   })


   test('debe tener la imagen igual al url y alt de los props', () => {

      const img = wrapper.find('img');

      /*enzyme ofrece el metodo .props() en el que se puede traer las properties de un elemento
      en forma de objeto:
         console.log(img.props().src);
      
      o si solo necesito una property del elemento
         console.log(img.prop('alt'));
      */
      expect(img.prop("src")).toBe(url);
      expect(img.prop("alt")).toBe(title);
   })
   

   test('debe tener la clase animate__bounce', () => {
      
      const div = wrapper.find('div');
      const className = div.prop('className');

      expect(className.includes('animate__bounce')).toBe(true);
   })


})

























