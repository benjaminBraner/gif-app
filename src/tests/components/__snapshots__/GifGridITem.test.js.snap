// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <p>
    some title
  </p>
  <img
    alt="some title"
    src="https://someurl"
  />
</div>
`;
